### DISCUSSION TITLE
revert cloudflare

### DISCUSSION THREAD

**Discussion by tttigo** [mozilla/hubs#3138] [Hubs Cloud Questions] [answered]
hey,

i've followed the instructions to use cloudflare, but when I saved it, all pages from my aws cloud stopped working. It only loads a white screen. How can I revert it, and what can i have done wrong?

thanks

**Comment by robinkwilson** [comment, marked as answer]
We'll need to delete the settings for Cloudflare in the Parameter Store:

1. Go to Parameter Store
    AWS console > Systems Manager > Parameter Store (try this shortcut link)
2. Check that you're in the correct region: Are your stack parameters listed in the menu?
    Yes, you're in the correct region
    No, select your deploy region (upper right hand corner next to your name)
3. Search for `workers.dev`, then delete those parameters
4. Search for `cdn` then delete that parameter
5. Check for any other parameters that relate to cloudflare
6. Restart your EC2 instances
    AWS console > EC2 > click on your HC server(s) then "Terminate" instances. New ones will start up immediately after.

Parameter store params to check against (thank you @aidenmiles for this in another discussion thread):
/ita/{stackname}/hubs/deploy/type
/keymaster/{stackname}/guardian-secret-key
/keymaster/{stackname}/janus-and-reticulum-admin
/keymaster/{stackname}/phoenix-key
/keymaster/{stackname}/postgrest-db-secret
/keymaster/{stackname}/reticulum-bot-access-key
/keymaster/{stackname}/reticulum-cookie
/keymaster/{stackname}/reticulum-oauth-token
/keymaster/{stackname}/send-email-password

**Reply by robinkwilson** [reply]
Did the two pages in the last step of the cloudflare setup guide render?

**Reply by tttigo** [reply]
can't find these parameters on parameter store. already checked the correct region.

**Reply by robinkwilson** [reply]
Do any of the above parameters match the params you have?

And are you still seeing a white screen? 
Can you look in your browser's developer tools > console + network tab and see any red errors?

**Reply by tttigo** [reply]
Yes![image](https://user-images.githubusercontent.com/6611885/95524670-5cbd5600-09a8-11eb-9399-67a4c54438c1.png)

Still seeing white screen.
![image](https://user-images.githubusercontent.com/6611885/95524828-d7867100-09a8-11eb-9fe8-cab6aa015bc4.png)

**Comment by robinkwilson** [comment]
Look at the **values** of the ones (probably the ones labelled cors_proxy_server) that do not match: 

/ita/{stackname}/hubs/deploy/type
/keymaster/{stackname}/guardian-secret-key
/keymaster/{stackname}/janus-and-reticulum-admin
/keymaster/{stackname}/phoenix-key
/keymaster/{stackname}/postgrest-db-secret
/keymaster/{stackname}/reticulum-bot-access-key
/keymaster/{stackname}/reticulum-cookie
/keymaster/{stackname}/reticulum-oauth-token
/keymaster/{stackname}/send-email-password

If the **values** look like it's the cloudflare workers you specified, delete them.

If I remember correctly, there are two that are not labelled cors_proxy.

**Reply by tttigo** [reply]
Sorry I'm confused. Should I delete all parameters except the ones you've listed?

**Reply by robinkwilson** [reply]
Only the ones with **values** that look like it has to do with your cloudflare workers.

To see the **values**, click on one parameter then click "Show".

**Reply by robinkwilson** [reply]
You may have additional parameters that you set in the Admin panel other than the above defaults, we want to keep those ones.

**Comment by tttigo** [comment]
Thanks I got it. It's now working. Now how can I set up Cloudflare correctly?

I followed the instructions and this happened.

"Verify your workers are working. This link should show the Mozilla homepage, and this link should should the Hubs "What's New" page." Its working, but when I check "Use Cloudflare Worker" and save: White Screen.

**Reply by robinkwilson** [reply]
Wonderful! Glad it's back! 

You can go through again, double check that the last two pages render before checking the checkbox. OR alternatively, you could leave it as is and continue using Cloudfront. We're working on deprecating cloudflare workers (no landing date yet on that though).

**Reply by tttigo** [reply]
I've repeated the process and got white screen again :(

**Reply by n8icus** [reply]
I just had this same problem redeploying my custom client seems to have fixed the issue.

**Reply by tttigo** [reply]
It worked redeploying my custom client, thanks!!

**Comment by n8icus** [comment]
Another note after I changed to Cloudflare. After re-deploying my custom client, everything seemed to be working just fine. The only issue I found is in Spoke. The image objects in the scene all load with the default broken link image. I noticed that image assets brought in from my assets now had the cloudflare worker proxy url for the first section of the url.  Changing the urls to match that makes everything work fine. The strange thing though is if I publish a scene without fixing those links, they show up fine in the rooms. So fixing the links only really makes it so that they display properly in spoke. Thought this might be helpful info. 
![image](https://user-images.githubusercontent.com/18128994/96039635-fbc5d000-0e36-11eb-90cc-6c0c12fd55dc.png)

**Comment by PaintedShepherd** [comment]
i had a white page also, but after redeploying it worked. We also needed to redeploy our spoke client to make it work. Thanks for your tips
